File: 349-intersection-of-two-arrays/intersection-of-two-arrays.py
def intersection(list1 : List[int] , list2 : List[int]) -> List[int]:
    return list(set(list1) & set(list2))
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        return intersection(nums1,nums2)

File: 349-intersection-of-two-arrays/intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> numSet(nums1.begin(), nums1.end());
        vector<int> res;

        for (int num : nums2) {
            if (numSet.count(num)) {
                res.push_back(num);
                numSet.erase(num); 
            }
        }

        return res;
    }
};
